### ISSUE TITLE
Module 2 Skip Logic or Saving Issue: MENSHIS2_1 and MENSHIS5

### ISSUE THREAD

**Issue by KELSEYDOWLING7** [episphere/questionnaire#440]
Based on the skip logic in the Qx document and QC rules, if MENSHISA is less than 0 or null, MENSHIS2_1 must be null. Same logic applies to MENSHIS5.

However these participants with null values were able to answer MENSHIS2_1:
4477623862 2025262984 5804126258 1454524180 7132444744 9621192890 6749884374 9880878735 9871441063 5195870819 8984301161 9236220854 9537726454 7928349908 2006727084 3497276532 9817027098 5585167438

And these participants with null values were able to answer MENSHIS5:
2025262984 5585167438 7928349908 1454524180 2261805341 9621192890 9579489549 9030288749 9871441063 9282536005 2907887692 1108663459 9880878735 9236220854 5342664636 5195870819 9537726454 7132444744 8984301161 2006727084 6749884374 3497276532 9817027098 7214144756 1223732780 4477623862

There is either an issue with the response value being saved for MENSHISA or a skip flow issue. I see a comment that the MENSHIS5 logic was changed last year, but it doesn't seem as though either question should be seen if MENSHISA is null. Their are cases where Module 2 was completed as recent as last week or as old as 2022, so it seems like these have been consistent issues.

**Comment by KELSEYDOWLING7** [comment]
I was unsure who in IMS to tag, as well as who is who by the usernames. Feel free to add or remove people tagged as needed

**Comment by boyd-mj** [comment]
You got everyone, thanks Kelsey!

**Comment by joshid-ims** [comment]
I could not reproduce the issue for MENSHIS2_1 but logic needs to be changed for MENSHIS5. 

Can you please tell what exactly was entered in case of MENSHIS2_1 issue?

**Comment by KELSEYDOWLING7** [comment]
In the case of MENSHIS2_1, if a participant does not answer with an age in MENSHISA, they should not be able to see MENSHIS2_1. For the participants listed above, they all saw and answered MENSHIS2_1 without a MENSHISA value entered

**Comment by joshid-ims** [comment]
Yea, that is what I am not able to reproduce. I tried both, selected 44 and also chose not to answer MENSHISA.

**Comment by KELSEYDOWLING7** [comment]
Ok that one might be a saving issue then. Meaning its possible they answered MENSHISA but we somehow don't have the data. 

@anthonypetersen is this something you or your team would be able to check? This may be linked to the data saving fail we discovered in this issue with Daniel: https://github.com/episphere/connect/issues/1110

**Comment by KELSEYDOWLING7** [comment]
Hmm I'm not sure. It looks like the markup and the QC rule are inconsistent then. "MENSHISA is less than 0 or null" could mean that MENSHIS_SCR=44 but not guaranteed.

**Comment by boyd-mj** [comment]
@KELSEYDOWLING7 could you check with your team and have Aileen make any changes to the questionnaire docs if content changes are needed? Thanks.

**Comment by KELSEYDOWLING7** [comment]
@cunnaneaq Do you mind reviewing this skip flow logic?

**Comment by KELSEYDOWLING7** [comment]
Me and Aileen reviewed. It looks like the rule for MENSHIS5 is following the old skip flow pattern (skip flow chart below) but the order of the questions have been changed. MENSHIS5 is now before MENSHIS9. We're thinking the QC rule may need to be updated accordingly.
![image](https://github.com/user-attachments/assets/1e001862-d8fd-40fa-a91e-4f5cab5a2b23)

**Comment by cunnaneaq** [comment]
I agree that MENSHISA = null with responses seen fro MENSHIS2_1 looks like a saving issue according to the qx doc and @joshid-ims unable to reproduce.

I looked back at an [old questionnaire document](https://nih.app.box.com/file/888095808900) which matches Kelsey's flow chart above.  Screenshots from the qx doc in prod are below

![Screenshot 2024-12-09 at 3 12 12 PM](https://github.com/user-attachments/assets/0e169364-e281-43f9-93b1-be5ddff58508)

![Screenshot 2024-12-09 at 3 10 48 PM](https://github.com/user-attachments/assets/1b65299b-6758-4e3e-a485-9d473ab532c4)

**Comment by KELSEYDOWLING7** [comment]
@cunnaneaq Circling back to this one. 

Looking at the most recent markdown, a null response to MENSHIS OR answering 44 to MENSHIS should prevent a participant from seeing MENSHIS2. These participants should have been taken to INTROMENSHIS2, which is now above MENSHIS9 and MENSHIS5.
![image](https://github.com/user-attachments/assets/fc154e1c-32d0-4ad0-8950-904858fa7975)


By that same logic, a participant would be taken to INTROMENSHIS2, and so long as they did not answer MENSHIS with 44, they would see MENSHIS5. 
![image](https://github.com/user-attachments/assets/c31576cb-7404-402b-96ac-ae6bc9021dc8)



My conclusion would be that:
1. A participant seeing MENSHIS5 after a null MENSHIS response is not an issue. Instead, the QC rule for MENSHIS5 needs to be updated to exclude only those who answered 44 to MENSHIS.
2. A participant seeing MENSHIS2 after a null MENSHIS response IS in fact an issue. They should not be seeing this question. However, as we don't plan to change the skip logic of Baseline modules, this may just have to be a known issue as remain as that.  We can keep the QC rule as it is and just capture these errors. This can be considered as additional unexpected data captured as we determined with some of the mod4 issues. 


Do you agree?

**Comment by KELSEYDOWLING7** [comment]
@anthonypetersen  Now that I think about it, this second issue below may possibly contribute to the MCS issues we were seeing last week.

"A participant seeing MENSHIS2 after a null MENSHIS response IS in fact an issue. They should not be seeing this question. However, as we don't plan to change the skip logic of Baseline modules, this may just have to be a known issue as remain as that. We can keep the QC rule as it is and just capture these errors. This can be considered as additional unexpected data captured as we determined with some of the mod4 issues."

**Comment by KELSEYDOWLING7** [comment]
@anthonypetersen  Please disregard, this is from Module 2. Similar variables, but I forgot that the MCS only pulls from the BUM/BU surveys menstrual history responses
